### DISCUSSION TITLE
calcODF pole figures not matching the original measured PF

### DISCUSSION THREAD

**Discussion by erescastellanos** [mtex-toolbox/mtex#1449] [Ask Anything]
Hi! 

I am using calcODF to compute my ODF and the obtained PF do not match the original measured ones. The original PF were incomplete, but we can still see a fair amount of them. This is an iron bcc structure (m3-m) and I am assuming no sample symmetry. 

I have tried using different flags to deactivate the ghost correction, calculate the Fourier series with several orders, etc - always same result. I would say that the problem seems to be the fact that PF are incomplete and MTEX reads them as if the values of the points in the outer part of them were zero, but I am not sure and I don't know how to solve it either.

Could you help me out with this? You can find the measured pole figure files and a comparison of the obtained and measured PF attached. The code I used is also below,

![PF_measured](https://user-images.githubusercontent.com/106779650/171747795-cdabc541-a25c-43cc-ac85-1aff13674a2e.PNG)
![PF_comparison](https://user-images.githubusercontent.com/106779650/171745523-0c6165b9-63be-45b0-96ec-65980ff1e4b1.png)
[04C3Si_211_Ferr.txt](https://github.com/mtex-toolbox/mtex/files/8828067/04C3Si_211_Ferr.txt)
[04C3Si_200_Ferr.txt](https://github.com/mtex-toolbox/mtex/files/8828068/04C3Si_200_Ferr.txt)
[04C3Si_110_Ferr.txt](https://github.com/mtex-toolbox/mtex/files/8828069/04C3Si_110_Ferr.txt)

close all, clear all
%% Import Script for PoleFigure Data
%
% This script was automatically created by the import wizard. You should
% run the whoole script or parts of it in order to import your data. There
% is no problem in making any changes to this script.

%% Specify Crystal and Specimen Symmetries

% crystal symmetry
CS = crystalSymmetry('m-3m', [1 1 1]);

% specimen symmetry
SS = specimenSymmetry('1');

% plotting convention
setMTEXpref('xAxisDirection','east');
setMTEXpref('zAxisDirection','outOfPlane');

%% Specify File Names

% path to files
pname = '.\';

% which files to be imported
fname{1} = {...
  [pname '04C3Si_110_Ferr.txt'],...
  [pname '04C3Si_200_Ferr.txt'],...
  [pname '04C3Si_211_Ferr.txt'],...
  };

%% Specify Miller Indice

h = { ...
  Miller(1,1,0,CS),...
  Miller(2,0,0,CS),...
  Miller(2,1,1,CS),...
  };

%% Import the Data

% create a Pole Figure variable containing the data
pf = loadPoleFigure(fname{1},h,CS,'interface','uxd');
odf_50C_prim = calcODF(pf,'halfwidth',5*degree, 'all');

figure()
plot(pf,'contourf')

figure()
plotPDF(odf_50C_prim,Miller(1,1,0,CS))
figure()
plotPDF(odf_50C_prim,Miller(2,0,0,CS))
figure()
plotPDF(odf_50C_prim,Miller(2,1,1,CS))

**Comment by kilir** [comment]
Hi,
did you use a defocussing and a background correction?
Cheers,
Rüdiger

**Comment by erescastellanos** [comment]
Hi!

so I don't have a background file right now, so I could not apply defocusing and background correction. Do you think it can affect that much?

I think that the problem is related to the fact that the original pole figures are incomplete, that's why the odf pole figures always end up having maximum intensity in their center. I guess that calcODF reads all the missing values as 0.

Thanks for your help!

Adriana Eres-Castellanos

________________________________
From: Rüdiger Kilian ***@***.***>
Sent: Friday, June 3, 2022, 9:38 AM
To: mtex-toolbox/mtex ***@***.***>
Cc: Adriana Eres Castellanos ***@***.***>; Author ***@***.***>
Subject: [External] Re: [mtex-toolbox/mtex] calcODF pole figures not matching the original measured PF (Discussion #1449)


Hi,
did you use a defocussing and a background correction?
Cheers,
Rüdiger

**Comment by erescastellanos** [comment]
Hi!

I just wanted to give an update on this thread, since I have managed to solve the problem and I think this may be useful for people encountering the same problem in the future. 

The problem was solved by increasing the resolution of the ODF to 10*degrees:

odf_50C_prim = calcODF(pf,'resolution',10*degree);

![new_PF](https://user-images.githubusercontent.com/106779650/171845822-159549ae-e9da-4ce4-ba28-92a4b7934738.png)

I guess the problem was also due to the low MRD intensity in the pole figures.
